homeassistant/exceptions.py

```diff
@@ -21,11 +24,42 @@ def __init__(
         translation_placeholders: dict[str, str] | None = None,
     ) -> None:
         """Initialize exception."""
+        if not args and translation_key and translation_domain:
+            self.generate_message = True
+            args = (translation_key,)
+
         super().__init__(*args)
         self.translation_domain = translation_domain
         self.translation_key = translation_key
         self.translation_placeholders = translation_placeholders
 
+    def __str__(self) -> str:
+        """Return exception message.
+
+        If no message was passed to `__init__`, the exception message is generated from
+        the translation_key. The message will be in English, regardless of the configured
+        language.
+        """
+
+        if self._message:
+            return self._message
+
+        if not self.generate_message:
+            self._message = super().__str__()
+            return self._message
+
+        if TYPE_CHECKING:
+            assert self.translation_key is not None
+            assert self.translation_domain is not None
+
+        # pylint: disable-next=import-outside-toplevel
+        from .helpers.translation import async_get_exception_message
```

Comment:
It would be nice to break this exception this out into a new module at some point so we can avoid the late import

Reply:
Followed up with https://github.com/home-assistant/core/pull/113683